modules/pxp-module-puppet

```diff
@@ -70,7 +70,7 @@ end
 
 def force_unicode(s)
   # Puppet still returns UTF-8 when locale is C or POSIX, so force encoding to that.
-  if s.encoding == Encoding::ASCII_8BIT
+  if s.encoding != Encoding::UTF_8
```

Comment:
The issue it's trying to fix is that SLES 10 and 11 (and a few other platforms) run services (i.e. pxp-agent) with a POSIX locale, which results in encoding coming back as `US_ASCII` (I mistakenly read it as `ASCII_8BIT` before).

However, Puppet's output can still include non-ASCII characters, resulting in an error thrown later when we try to use a regex.